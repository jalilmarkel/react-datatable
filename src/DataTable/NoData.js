import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const NoDataWrapperMessage = styled.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: 16px;
`;

const NoDataWrapper = ({ component }) => (
  <NoDataWrapperMessage>
    {component}
  </NoDataWrapperMessage>
);

NoDataWrapper.propTypes = {
  component: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
};

export default NoDataWrapper;
