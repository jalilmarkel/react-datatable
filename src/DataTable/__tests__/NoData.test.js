import 'jest-styled-components';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import NoData from '../NoData';

afterEach(cleanup);

test('should render correctly when a component is passed that is a string', () => {
  const { container } = render(<NoData component="A Component that is passed in">A String that is passed in</NoData>);
  expect(container.firstChild).toMatchSnapshot();
});

test('should render correctly when a component is passed that is a react component', () => {
  const { container } = render(<NoData component={<div>A String that is passed in</div>}><div>A String that is passed in</div></NoData>);
  expect(container.firstChild).toMatchSnapshot();
});
