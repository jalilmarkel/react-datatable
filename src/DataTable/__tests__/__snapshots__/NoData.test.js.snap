// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when a component is passed that is a react component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: 16px;
}

<div
  class="c0"
>
  <div>
    A String that is passed in
  </div>
</div>
`;

exports[`should render correctly when a component is passed that is a string 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: 16px;
}

<div
  class="c0"
>
  A Component that is passed in
</div>
`;
